File: server/handlers/bountyHunting.py
from flask import render_template
from server.powers import get_system_power_info, is_system_anarchy
from server.tasks.bountyHunting import has_stations
from server.tasks.tasks import TaskDescription, getTaskType, is_task_own_strength, isPowersWeakness, isTaskACrime, systemNotes


def handle_bounty_hunting(request, database):
    system = request.args.get("system")
    task = request.args.get("taskName")
    power = request.args.get("power")

    stations = has_stations(system, database)
    if (stations == -1):
        stations = "No"
    
    powerInfo = get_system_power_info(system, database)
    systemState = powerInfo[0]
    controllingPower = powerInfo[1]
    if is_system_anarchy(system, database):
        #bounty hunting dosen't work in anarchy!
        return render_template(
            "does_not_work.html",
            ERRORDATA="Bounty hunting cannot be completed in anarchy systems",
            ERRORCODE="INCOMPLETABLE"
            )
    
    
    return render_template(
            "tasks/general.html",
            system=system,
            power=power,
            currentPower=controllingPower,
            currentState=systemState,
            isAnarchy="YES" if is_system_anarchy(system, database) else "NO",
            taskName=task,
            taskType=getTaskType(task),
            isIllegal="Is" if isTaskACrime(task, is_system_anarchy(system, database)) else "isn't",
            isOpposingWeakness=isPowersWeakness(power, task),
            taskDescription=TaskDescription(task, power, system, powerInfo, database),
            systemNotes=systemNotes(power, system, database),
            isOwnStrength="is" if is_task_own_strength(task, controllingPower) else "isn't"
        )

File: src/handlers/tasks/bountyHunting.go
package tasks

import (
	"github.com/labstack/echo/v5"
	"niceygy.net/powerplay-assistant/src/database"
	"niceygy.net/powerplay-assistant/src/utils"
)

const bounty_task_code string = "BTHT"

func HandleBountyHunting(c *echo.Context) error {
	system := c.QueryParam("system")
	power := c.QueryParam("power")

	if database.IsSystemAnarchy(system) {
		return c.HTML(200, utils.RenderTemplate("templates/does_not_work.html", map[any]any{
			"ERRORDATA": "Bounty hunting cannot be completed in anarchy systems.",
			"ERRORCODE": "INCOMPLETABLE",
		}))
	} else {
		user_power_shortcode := database.PowerFullToShort(power)
		state, controlling_power := database.GetSystemPowerInfo(system)
		return c.HTML(200, utils.RenderTemplate("templates/tasks/general.html", map[any]any{
			"system":             system,
			"power":              power,
			"currentPower":       controlling_power,
			"currentState":       state,
			"isAnarchy":          "NO",
			"taskName":           "Bounty Hunting",
			"isIlligal":          "NO",
			"isOpposingWeakness": utils.IsOpposingWeakness(bounty_task_code, user_power_shortcode),
			"taskDescription":    utils.TASKDESCRIPTIONS[bounty_task_code],
			"systemNotes":        database.GetExtraInfo(system),
			"isOwnStrength":      utils.IsOwnStrength(bounty_task_code, user_power_shortcode),
		}))
	}
}
